services/users/profile_picture.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package users
+
+import (
+	"bytes"
+	"hash/fnv"
+	"image"
+	"image/color"
+	"image/draw"
+	"image/png"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/freetype"
+	"github.com/golang/freetype/truetype"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/utils/fileutils"
+)
+
+const (
+	ImageProfilePixelDimension = 128
+)
+
+func (us *UserService) GetDefaultProfileImage(user *model.User) ([]byte, error) {
+	if user.IsBot {
+		return model.BotDefaultImage, nil
+	}
+
+	return createProfileImage(user.Username, user.Id, *us.config().FileSettings.InitialFont)
+}
+
+func createProfileImage(username string, userID string, initialFont string) ([]byte, error) {
```

Comment:
Couldn't quite pull the trigger on refactoring this one and now it found new life in your package :grimacing:

Reply:
Ehe, it can be still refactored I just didn't wanted to go into it.